Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp

```diff
@@ -0,0 +1,132 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "Acts/Utilities/Result.hpp"
+#include "ActsExamples/EventData/PropagationSummary.hpp"
+#include "ActsExamples/Propagation/PropagationAlgorithm.hpp"
+#include "ActsExamples/Propagation/PropagatorInterface.hpp"
+#include "ActsExamples/Traccc/DetrayStore.hpp"
+
+#include <detray/navigation/navigator.hpp>
+#include <detray/utils/inspectors.hpp>
+
+namespace ActsExamples {
+
+/// Define the algebra type
+using DetrayAlgebraType = typename Acts::DetrayHostDetector::algebra_type;
```

Comment:
This will be followed up in another PR, as discussed - I will move the DetrayPropagator to create a `detray::propagator` for each track - so this will change as well the how I can deduce the inspectors.